### PR TITLE
#5222 Remove `ApplicationName` and `ApplicationVersion` config parameters

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

This PR removes unnecessary configuration parametes from `cardano-node` config.
https://github.com/input-output-hk/cardano-node/issues/5222
Requires https://github.com/input-output-hk/cardano-api/pull/8
see also: https://input-output-rnd.slack.com/archives/CCRB7BU8Y/p1683739407050779

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [x] CI passes. See note on CI.  The following CI checks are required:
  - [x] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [x] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [x] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [x] Self-reviewed the diff

### CODE REVIEW

File: .github/workflows/check-mainnet-config.yml [hunk 1]
```diff
@@ -33,21 +33,52 @@ jobs:
...
-        copyFile "mainnet-conway-genesis.json"
-        copyFile "mainnet-config.json"
-        copyFile "mainnet-shelley-genesis.json"
-        copyFile "mainnet-topology.json"
-
-    - name: Check mainnet for mainnet configuration differences
+
+    - name: Check if configuration from nix is a superset of in-repo configuration
+      shell: python
       run: |
```
Comment (carbolymer) [inline .github/workflows/check-mainnet-config.yml:39]: This now checks if all configuration fields from repo configuration are present in the nix configuration and are equal. Extra fields in nix configuration do not make this check fail.

**Review by Jimbo4350 — APPROVED** [review]

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
https://github.com/input-output-hk/cardano-node/pull/5240#discussion_r1206820184

File: .github/workflows/check-mainnet-config.yml (hunk 1 above)
Comment (carbolymer) [reply .github/workflows/check-mainnet-config.yml:38]: We have a couple of python snippets in `cardano-node` already.
Anyway, I've rewritten this using `jq` - I hope it's better now.

**Review by Jimbo4350 — APPROVED** [review]

**Review by CarlosLopezDeLara — APPROVED** [review]
Nice!

**Comment by carbolymer** [comment]
Hydra build failures are unrelated to this PR, they will be fixed by: https://github.com/input-output-hk/cardano-node/pull/5274/files

**Comment by angerman** [comment]
What happens if those keys _are_ still in the config file? Does this produce a sensible error message, or simply ignore them?

**Comment by Jimbo4350** [comment]
> What happens if those keys _are_ still in the config file? Does this produce a sensible error message, or simply ignore them?

They are ignored. No need to produce an error message because they are legacy Byron parameters.

**Review by disassembler — APPROVED** [review]

**Pull request merged**